zcash_primitives/src/transaction/sighash.rs

```diff
@@ -127,7 +104,7 @@ fn joinsplits_hash(
 ) -> Blake2bHash {
     let mut data = Vec::with_capacity(
         joinsplits.len()
-            * if txversion.uses_groth_proofs() {
+            * if txversion.has_sapling() {
```

Comment:
Per my earlier note, this is switching Sprout logic on the presence of Sapling logic, which feels brittle.

Reply:
I'm going to see what it takes to plumb in the consensus branch id here instead.